contracts/Pool.sol

```diff
@@ -529,3 +529,105 @@ library ReceiverWeightsImpl {
         }
     }
 }
+
+/// @notice A list of cycles and their deltas of amounts received by a proxy.
+/// For each cycle there are stored two deltas, one for the cycle itself
+/// and one for the cycle right after it.
+/// It reduces storage access for some usage scenarios.
+/// The cycle is described by its own entry and an entry for the previous cycle.
+/// Iterable and with random access.
+struct ProxyDeltas {
+    mapping(uint64 => ProxyDeltasImpl.ProxyDeltaStored) data;
+}
+
+/// @notice Helper methods for proxy deltas list.
+/// The list works optimally if after applying a series of changes it's iterated over.
+/// The list uses 2 words of storage per stored cycle.
+library ProxyDeltasImpl {
+    struct ProxyDeltaStored {
+        uint64 next;
+        // --- SLOT BOUNDARY
+        Pool.AmtDelta delta;
+    }
+
+    uint64 internal constant CYCLE_ROOT = 0;
+    uint64 internal constant CYCLE_UNINITIALIZED = 0;
```

Comment:
For me, the code would become more readable if we would use only one name for the same constant value. Otherwise one might be lead to believe that this are different values.

But this could also be a follow-up where we make the same change to to `ReceiverWeights`.

Reply:
I kind of agree. I don't like the fact that the cycle number `0` has 2 different meanings depending on context, but you're right that having 2 names for the same value is madness. And no, they can't have different values, `CYCLE_ROOT` must equal `CYCLE_UNINITIALIZED`, which must equal `0`. Fixed.